src/Chapter3.hs

```diff
@@ -560,22 +634,42 @@ introducing extra newtypes.
 🕯 HINT: if you complete this task properly, you don't need to change the
     implementation of the "hitPlayer" function at all!
 -}
+newtype Health = Health {
+  unHealth :: Int
+}
+
+newtype Armor = Armor {
+  unArmor :: Int
+}
+
+newtype Attack = Attack {
+  unAttack :: Int
+}
+
+newtype Dexterity = Dexterity {
+  unDexterity :: Int
+}
+
+newtype Strength = Strength {
+  unStrength :: Int
+}
+
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
+    { playerHealth    :: Health
+    , playerArmor     :: Armor
+    , playerAttack    :: Attack
+    , playerDexterity :: Dexterity
+    , playerStrength  :: Strength
     }
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
+calculatePlayerDamage :: Attack -> Strength -> Int
+calculatePlayerDamage attack strength = unAttack attack + unStrength strength
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDefense :: Armor -> Dexterity -> Int
+calculatePlayerDefense armor dexterity = unArmor armor * unDexterity dexterity
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerHit :: Int -> Int -> Health -> Health
+calculatePlayerHit damage defense health = Health (unHealth health + defense - damage)
```

Comment:
I feel like more newtypes can be used here:

1. You can introduce the `Damage` newtype and use it it `calculatePlayerDamage` and `calculatePlayerHit`
2. The `calculatePlayerDefense` function can return a value of type `Armor` as well

Nevertheless, great work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype